### ISSUE TITLE
Pokemon Moves

### ISSUE THREAD

**Issue by vincentchengws** [PocketMobsters/PokeMobBot#600]
Is there any function to sort pokemon according to the dps of their moves?
Then keep the pokemon with best moves and transfer the rest.

Since moves are more important than IV and CP.

What do you guys think?

**Comment by jonsnowwatch** [comment]
no, moves are not more important than IV. Plus, just because some move set yields "best dps", doesnt mean they are the best against all pokemons in all situations.

**Comment by vincentchengws** [comment]
But Why don't we select moves matching the pokemon. ie, Draginite, we select moves with dragon ability. But don't select steel wing.

**Comment by vincentchengws** [comment]
So, when we face different pokemon, we select different pokemon to counter each other.
And at last, all your pokemon will have highest dps. against a specific type of pokemon

**Comment by CptnKvin** [comment]
imagine you fight with a water type pokemon against a water type pokemon.
obviously you won't need stab attacks

**Comment by nehirakdag** [comment]
Moves arent more important than IV or CP, but it definitely would not hurt to have that option either.

Since the bot encounters masses of each species and keeps only a few, I'd want to be able to keep, say above 90% IV of that pokemon with the highest DPS moveset rather than a 98% IV with terrible attacks. Believe me, it's annoying to work around things when the bot keeps a 96% Arcanine that knows the 10.29dps Bulldoze attack and wants to transfer a 93% one that knows Fire Blast(24.39 dps).

I take the question as an innocent improvement suggestion, no? It would certainly make sense to add them to the bot since moves and their DPS are two very crucial parts of the game.
(List of moves with their DPS in case anyone is interested: [https://thesilphroad.com/research#moves](https://thesilphroad.com/research#moves))

**Comment by BrttClne22** [comment]
I'm not sure how long they will actively make changes to moves but that is something to take into account. IVs aren't going anywhere... That top notch Vaporeon you had two weeks ago with it's top moveset is now significantly weaker. Not saying that moves changing will always be the case, but I wouldn't collect based on them.

**Comment by vincentchengws** [comment]
@nehirakdag Yes exactly. You get my point :) it is a suggestion for future improvement :)

@BrttClne22 , why is the vaporeon significantly weaker now?

**Comment by BrttClne22** [comment]
@vincentchengws They made some pretty drastic changes on move DPS... Last week or so? I can't find a full list ATM but TMartn covered the changes on his YouTube channel and there's a reference in the article below. Sure theres a full list somewhere. I'll post if I find it. 

"Significantly" may be an overstatement, but you get the point.

http://www.vg247.com/2016/07/30/pokemon-go-update-seems-to-have-made-adjustments-to-more-pokemon-than-just-vaporeon/

**Comment by nehirakdag** [comment]
@BrttClne22 yes they nerfed Vaporeon, but as far as I know the best moveset that a Vaporeon can have remains the same, right? I'm sad over how my Vaporeon's  Hydro Pump got reduced from say 26dps to 23.68dps, but I'm certainly not going to wish it knew Aqua tail (19.15dps) all of a sudden because of this change either.

I'm saying we could be given a CHOICE that complements IVs, not a standalone deciding factor.

**Comment by cl3537git** [comment]
I agree with the OP, we should be able to add to the Transfer Filter Moveset as well.
I have a 100IV Lapras with Ice Shard at 2563cp and it is a lousy move and really slow. The bot transferred away my 95 - 99 IV Laprii with better movesets and the one it kept is useless to me for attacking.

The transfer logic should cascade down and we should be able to priortize transfer criteria min keep critieria:

Moveset: Blizzard
IV: 100
CP: 1000

Because if I just set min keep criteria to 100IV and Duplicates to 2 or 3 I might lose a 99IV Lapras with the moveset I want.
